tcms_tap_plugin/__init__.py

```diff
@@ -14,8 +14,25 @@ def __init__(self):
     def parse(self, tap_file, progress_cb=None):
         self.backend.configure()
 
+        before_first_test = True
+        test_case_run_id = False
+        status_id = False
+        trace_back = []
+        comment = ''
         for line in Parser().parse_file(tap_file):
-            if not isinstance(line, Result):
+            if isinstance(line, Result):
+                # Before parse result line, we need to add
+                # traceback as comment to the previous result
+                if test_case_run_id and status_id and trace_back:
+                    comment += '\n' + '\n'.join(trace_back)
+                    self.backend.update_test_case_run(
+                        test_case_run_id, status_id, comment)
+                trace_back = []
+                before_first_test = False
+            elif isinstance(line, Diagnostic):
+                if not before_first_test:
+                    trace_back.append(line.text[2:])
```

Comment:
If it is a `Diagnostic` message why not just append the line to `trace_back` and use it/reset it whenever we see fit? I don't see the need for yet another variable and if condition.